OrbitCore/LinuxEventTracer.h

```diff
@@ -5,46 +5,132 @@
+
+#if 0
+class LinuxFunction {
+ public:
+  LinuxFunction(std::string binary_path,
+                uint64_t file_offset,
+                uint64_t virtual_address)
+      : binary_path_{std::move(binary_path)},
+        file_offset_{file_offset},
+        virtual_address_{virtual_address} {}
+
+  const std::string& BinaryPath() const {
+    return binary_path_;
+  }
+
+  uint64_t FileOffset() const {
+    return file_offset_;
+  }
+
+  uint64_t VirtualAddress() const {
+    return virtual_address_;
+  }
+
+ private:
+  std::string binary_path_;
+  uint64_t file_offset_;
+  uint64_t virtual_address_;
+};
+#endif
+
+class LinuxEventTracerThread {
+ public:
+  LinuxEventTracerThread(pid_t pid, double frequency,
+                         std::vector<Function*> functions)
+      : pid_(pid), frequency_(frequency), functions_(std::move(functions)) {}
+
+  LinuxEventTracerThread(const LinuxEventTracerThread&) = delete;
+  LinuxEventTracerThread& operator=(const LinuxEventTracerThread&) = delete;
+  LinuxEventTracerThread(LinuxEventTracerThread&&) = default;
+  LinuxEventTracerThread& operator=(LinuxEventTracerThread&&) = default;
+
+  void Run(const std::shared_ptr<std::atomic<bool>>& a_ExitRequested);
+
+ private:
+  static constexpr int32_t ROUND_ROBIN_BATCH_SIZE = 5;
+
+  pid_t pid_;
+  double frequency_;
+  std::vector<Function*> functions_;
+
+  uint64_t period_ns_ = 1000000;
+  int32_t cpus_ = 0;
+
+  bool ComputeFrequency();
+  void LoadNumCpus();
+  void HandleCallstack(pid_t tid, uint64_t timestamp,
+                       const std::vector<unwindstack::FrameData>& frames);
+};
 
-//-----------------------------------------------------------------------------
 class LinuxEventTracer {
  public:
-  static const uint32_t ROUND_ROBIN_BATCH_SIZE = 5;
-  using Callback = std::function<void(const std::string& a_Data)>;
+  static constexpr double DEFAULT_SAMPLING_FREQUENCY = 1000.0;
+
+  LinuxEventTracer(pid_t pid, double frequency,
+                   std::vector<Function*> functions)
+      : pid_{pid}, frequency_{frequency}, functions_{std::move(functions)} {}
 
-  explicit LinuxEventTracer(pid_t a_Pid, uint32_t a_Frequency = 1000)
-      : m_Pid{a_Pid}, m_Frequency{a_Frequency} {}
+  explicit LinuxEventTracer(pid_t pid)
+      : LinuxEventTracer{pid, DEFAULT_SAMPLING_FREQUENCY, {}} {}
 
-  void Start();
-  void Stop();
-  bool IsRunning() const { return !(*m_ExitRequested); }
+  LinuxEventTracer(pid_t pid, double frequency)
+      : LinuxEventTracer{pid, frequency, {}} {}
 
- protected:
-  void CommandCallback(std::string a_Line);
+  LinuxEventTracer(pid_t pid, std::vector<Function*> functions)
+      : LinuxEventTracer{pid, DEFAULT_SAMPLING_FREQUENCY,
+                         std::move(functions)} {}
+
+  LinuxEventTracer(const LinuxEventTracer&) = delete;
+  LinuxEventTracer& operator=(const LinuxEventTracer&) = delete;
+  LinuxEventTracer(LinuxEventTracer&&) = default;
+  LinuxEventTracer& operator=(LinuxEventTracer&&) = default;
```

Comment:
same here - default constructor... and there are couple of places below.

Reply:
Same as above.